docs/developer/architecture/core/uisettings-service.asciidoc

```diff
@@ -38,3 +95,68 @@ export class MyPlugin implements Plugin {
 }
 
 ----
+
+[[uisettings-migrations]]
+=== Migrations
+
+[IMPORTANT]
+==============================================
+Migrations for 3rd party plugin advanced settings are not currently supported. If a 3rd party plugin registers an advanced setting, the setting is essentially permanent and cannot be fixed without manual intervention.
```

Comment:
Explicitly point out that migrations for 3rd party plugin advanced settings aren't supported right now.

Reply:
👍 LGTM. Could consider linking to the GH issue, but I don't think it's critical.